stdlib/messaging/rabbitmq/src/main/ballerina/src/rabbitmq/message.bal

```diff
@@ -14,62 +14,160 @@
 // specific language governing permissions and limitations
 // under the License.
 
-# Public Ballerina API - Ballerina RabbitMQ Message.
+import ballerinax/java;
+
+# Provides the functionality to manipulate the messages received by the consumer services.
 public type Message client object {
+   handle amqpChannel = JAVA_NULL;
+   private byte[] messageContent = [];
+   private int deliveryTag = -1;
+   private BasicProperties? properties = ();
+   private boolean ackStatus = false;
+   private boolean autoAck = true;
 
    # Acknowledge one or several received messages.
    #
    # + multiple - `true` to acknowledge all messages up to and including the message called on,
    #                `false` to acknowledge just the message called on.
    # + return - An error if an I/O error is encountered or nil if successful.
-   public remote function basicAck(boolean? multiple = false) returns Error? = external;
+   public remote function basicAck(boolean multiple = false) returns Error? {
+        var result = nativeBasicAck(self.amqpChannel, self.deliveryTag, multiple, self.autoAck, self.ackStatus);
+        self.ackStatus = true;
+        return result;
+   }
 
    # Reject one or several received messages.
    #
    # + multiple - `true` to reject all messages up to and including the message called on;
    #                `false` to reject just the message called on.
-   # + requeue - `true` if the rejected message(s) should be requeued rather than discarded/dead-lettered.
+   # + requeue - `true` if the rejected message(s) should be re-queued rather than discarded/dead-lettered.
    # + return - An error if an I/O error is encountered or nil if successful.
-   public remote function basicNack(boolean? multiple = false, boolean? requeue = true)
-                        returns Error? = external;
+   public remote function basicNack(boolean multiple = false, boolean requeue = true)
+                            returns Error? {
+        var result = nativeBasicNack(self.amqpChannel, self.deliveryTag, self.autoAck, self.ackStatus,
+                                multiple, requeue);
+        self.ackStatus = true;
+        return result;
+   }
 
    # Retrieves the delivery tag of the message.
    #
    # + return - int containing the delivery tag of the message.
-   public function getDeliveryTag() returns @tainted int = external;
+   public function getDeliveryTag() returns @tainted int {
+        if (self.deliveryTag > -1) {
+            return self.deliveryTag;
+        } else {
+            Error e = Error(message = "Delivery tag not properly initiliazed.");
+            panic e;
```

Comment:
Why this panicking instead returning the error?

Reply:
This is a situation which cannot occur, delivery tag is set at Message object creation. I had checked it just to be safe. If it comes inside the else clause it means that Message object creation failed at native code level. If this check seems unnecessary we'll just remove it.

Reply:
No let's keep this.